stories/__tests__/SQFormDialog.stories.test.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import {fireEvent, render, screen, waitFor} from '@testing-library/react';
+import {composeStories} from '@storybook/testing-react';
+
+import userEvent, {specialChars} from '@testing-library/user-event';
+import * as stories from '../SQFormDialog.stories';
+
+const {Default, WithValidation, WithAutoFocus} = composeStories(stories);
+
+window.alert = jest.fn();
+const handleClose = jest.fn();
+const handleSave = jest.fn();
+
+afterEach(() => {
+  window.alert.mockClear();
+  handleClose.mockClear();
+  handleSave.mockClear();
+});
+
+const mockData = {
+  hello: 'howdy'
+};
+
+describe('Tests for Default', () => {
+  it('should render a form dialog with a title', async () => {
+    const dialogTitleValue = 'Test';
+
+    render(
+      <Default
+        isOpen={true}
+        onSave={handleSave}
+        onClose={handleClose}
+        title={dialogTitleValue}
+      />
+    );
+
+    const dialogTitle = screen.getByText(dialogTitleValue);
+    expect(dialogTitle.textContent).toEqual(dialogTitleValue);
+  });
+
+  it('should handle form updates and submit said updates on save button click', async () => {
+    render(<Default isOpen={true} onSave={handleSave} onClose={handleClose} />);
+
+    const textField = screen.getByLabelText(/hello/i);
+    userEvent.type(textField, mockData.hello);
+
+    const saveButton = screen.getByRole('button', {name: /save/i});
+    userEvent.click(saveButton);
+
+    await waitFor(() => {
+      expect(handleSave).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  it('should handle form updates and submit values on `enter` keydown', async () => {
+    render(<Default isOpen={true} onSave={handleSave} onClose={handleClose} />);
+
+    const textField = screen.getByLabelText(/hello/i);
+    userEvent.type(textField, mockData.hello);
+
+    // https://testing-library.com/docs/ecosystem-user-event/#specialchars
+    userEvent.type(textField, `${specialChars.enter}`);
```

Comment:
You can save an import and simplify this to:
```js
userEvent.type(textField, '{enter}');
```
It will automatically recognize that as the `enter` key. They should probably update that in the `testing-library.com` docs

https://github.com/testing-library/user-event#special-characters